cmake/modules/SIMDExt.cmake

```diff
@@ -77,6 +77,11 @@ elseif(CMAKE_SYSTEM_PROCESSOR MATCHES "i386|i686|amd64|x86_64|AMD64")
       if(HAVE_INTEL_SSE4_2)
         set(SIMD_COMPILE_FLAGS "${SIMD_COMPILE_FLAGS} -msse4.2")
       endif()
+      CHECK_C_COMPILER_FLAG(-mavx2 HAVE_INTEL_AVX2)
```

Comment:
please note, the compiler on building host capable to emit avx2 instructions does not imply that the building host as the target platform also is able to execute avx2 instructions. you might want to *try to run* avx2 instructions on this platform beforehand to see if it's able to execute avx2.

or you could use function mutliversioning to choose the optimum implementation at runtime. see https://gcc.gnu.org/wiki/FunctionMultiVersioning

Reply:
@tchaikov 
It's true.
There's one file intel.c in this PR is used to detect whether the running hardware supports AVX2 instruction.

Reply:
ahh, sorry. i missed that part.